### PR TITLE
Suppress MISRA violations for POSIX functions in transport plaintext

### PR DESCRIPTION
**Pull request by aggarw13**
Suppress MISRA violations with annotations for use of `FD_SET` and `FD_ZERO` macros in `plaintext_posix.c. These are POSIX macros specified here: https://pubs.opengroup.org/onlinepubs/009695399/basedefs/sys/select.h.html

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

**Review by yourslab — DISMISSED** [review]

**Review by yourslab — APPROVED** [review]

File: platform/posix/transport/src/plaintext_posix.c [hunk 1]
```diff
@@ -170,7 +200,29 @@ int32_t Plaintext_Send( const NetworkContext_t * pNetworkContext,
         sendTimeout.tv_usec = 0;
     }
 
+    /* MISRA Directive 4.6 flags the following line for a violation of using a
+     * basic type "int" rather than a type that includes size and signedness information.
+     * We suppress the violation as the flagged type, "fd_set", is a POSIX
+     * system-specific type, and is used for the call to "select()". */
+    /* coverity[misra_c_2012_directive_4_6_violation] */
+    /* coverity[misra_c_2012_rule_14_4_violation] */
```
Comment (leegeth) [inline platform/posix/transport/src/plaintext_posix.c:208]: please add the explanation about why 14.4 is marked as ignored here as well

File: platform/posix/transport/src/plaintext_posix.c [hunk 2]
```diff
@@ -96,8 +96,38 @@ int32_t Plaintext_Recv( const NetworkContext_t * pNetworkContext,
...
     FD_ZERO( &readfds );
+
+    /* MISRA Directive 4.6 flags the following line for a violation of using a
+     * basic type "int" rather than a type that includes size and signedness information.
+     * We suppress the violation as the flagged type, "fd_set", is a POSIX
+     * system-specific type, and is used for the call to "select()". */
+    /* coverity[misra_c_2012_directive_4_6_violation] */
+
+    /* MISRA Directive 4.6, Rule 10.1 and Rules 13.4 flag the following line for
```
Comment (leegeth) [inline platform/posix/transport/src/plaintext_posix.c:119]: is this 4.6 or 10.8? Same for all the other places.

**Review by leegeth — CHANGES_REQUESTED** [review]
Please see the comments.

File: platform/posix/transport/src/plaintext_posix.c (hunk 1 above)
Comment (aggarw13) [reply platform/posix/transport/src/plaintext_posix.c:214]: Thanks for catching. Added.

**Review by leegeth — APPROVED** [review]

File: platform/posix/transport/src/plaintext_posix.c [hunk 3]
```diff
@@ -96,8 +96,38 @@ int32_t Plaintext_Recv( const NetworkContext_t * pNetworkContext,
...
+     * implementation of the "FD_SET()" POSIX macro. We suppress these violations
+     * "FD_SET" is a POSIX specific macro utility whose implementation
+     * is supplied by the system.
+     * The "FD_SET" macro is used as specified by the POSIX manual here:
+     * https://pubs.opengroup.org/onlinepubs/009695399/basedefs/sys/select.h.html */
+    /* coverity[misra_c_2012_directive_4_6_violation] */
+    /* coverity[misra_c_2012_rule_10_1_violation] */
+    /* coverity[misra_c_2012_rule_13_4_violation] */
+    /* coverity[misra_c_2012_rule_10_8_violation] */
+    FD_SETpNetworkConSET->socketDescriptor, &readfds );
```
Comment (leegeth) [inline platform/posix/transport/src/plaintext_posix.c:129]: compilation failure on this line.

**Review by leegeth — CHANGES_REQUESTED** [review]
compilation issue.

File: platform/posix/transport/src/plaintext_posix.c (hunk 3 above)
Comment (aggarw13) [reply platform/posix/transport/src/plaintext_posix.c]: Fixed

**Review by leegeth — APPROVED** [review]

**Review by dan4thewin — APPROVED** [review]

**Pull request merged**